src/protocols/bdx/TransferFacilitator.cpp

```diff
@@ -112,6 +112,7 @@ CHIP_ERROR Responder::PrepareForTransfer(System::Layer * layer, TransferRole rol
 void Responder::ResetTransfer()
 {
     mTransfer.Reset();
+    mStopPolling = true;
```

Comment:
Please get someone familiar with the src/protocols/bdx code to review this part?  That said, this is very suspect, since it's backing out part of #23642.  Can you explain why that change was wrong and why this is not re-introducing whatever issue that PR was fixing?

Reply:
@carol-apple Could you please take a look at this?

Reply:
Discussed with Carol and looks like there could be an issue with this not stopping polling. I need to debug this.

Reply:
So the fix for https://github.com/project-chip/connectedhomeip/pull/23642 was to address https://github.com/project-chip/connectedhomeip/issues/23573. It was also dealing with a negative cases. @nivi-apple will you please validate if moving the stop polling code back will also address https://github.com/project-chip/connectedhomeip/issues/23573?

cc: @harsha-rajendran

Reply:
Tested this by hacking the ota-requestor code to fail the first time after a successful query image and fail the software version check and call the code that rejects the transfer. on the second retry ota requestor queries image. provider was reset correctly with this fix and BDX continued successfully between the two.

Reply:
actually scratch the previous comment. I tested with the ota-provider app and the stop polling code here does break the use case for https://github.com/project-chip/connectedhomeip/pull/23642. Either we pull out this part of the change and test the part where bdx init doesn't come, we do send the status report out. if thats sufficient to stop polling we are good. @bzbarsky-apple what do you think?

Reply:
Should `mStopPolling = false` be added to `Responder::PrepareForTransfer`?

Reply:
Addressed Damian's comment and it works for issue 23573